initialize.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -e
+
+# Check if docker is installed
+if ! [ -x "$(command -v docker)" ]; then
+  echo 'Error: docker is not installed.' >&2
+  # Ask if user wants to install docker
+  read -p "Do you want to install docker? [y/n] " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    # Install docker
+    curl -fsSL https://get.docker.com -o get-docker.sh
+    sh get-docker.sh
+    # Check if docker is installed
+    if ! [ -x "$(command -v docker)" ]; then
+      echo 'Error: Could not install docker.' >&2
+      exit 1
+    fi
+  else
+    exit 1
+  fi
+fi
```

Comment:
is it also possible to write something if the requirements are actually installed?

Example:
"checking docker installation...."

then

"Docker is correctly installed with version X"

otherwise

what you have already done

Reply:
We should also check the version of docker and tell them that the version is too old in case. No need to manage the uninistall/reinstall because too complicated imho. But at least the users know the problem and can fix it on their own